### PR TITLE
(core|cli) enable EventBridge notifications; remove CloudTrail; update CDK version

### PR DESCRIPTION
**Pull request by kukushking**
### Feature or Bugfix

- Feature

### Detail
- Add event_bridge_enabled to bucket factory
- Remove CloudTrail from S3EventStage
- Update event patterns on S3EventStage & Kinesis stage
- Update CDK version

### Relates
- N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/aws_ddk_core/resources/_s3.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class BucketSchema(BaseSchema):
     versioned = fields.Bool(load_default=True)
     auto_delete_objects = fields.Bool()
     enforce_ssl = fields.Bool(load_default=True)
+    event_bridge_enabled = fields.Bool(load_default=True)
```
Comment (jaidisido) [inline core/aws_ddk_core/resources/_s3.py:34]: Do you think users will actually want to enable this differently from one env to the other?

**Review by jaidisido — APPROVED** [review]
Left a minor comment

File: core/aws_ddk_core/resources/_s3.py (hunk 1 above)
Comment (kukushking) [reply core/aws_ddk_core/resources/_s3.py:34]: Good point, unlikely -- I'll remove it

File: core/aws_ddk_core/stages/s3_event.py [hunk 2]
```diff
@@ -53,53 +50,31 @@ def __init__(
...
-                include_global_service_events=False,
-            )
-            if not cloudtrail_trail
-            else cloudtrail_trail
-        )
-        self._trail.add_s3_event_selector(
-            s3_selector=[S3EventSelector(bucket=self._bucket, object_prefix=key_prefix)],
-            include_management_events=False,
+            detail=detail,
+            detail_type=["Object Created"],
```
Comment (jaidisido) [inline core/aws_ddk_core/stages/s3_event.py:77]: Shouldn't this be `event_names`?

File: core/aws_ddk_core/stages/kinesis_s3.py [hunk 3]
```diff
@@ -170,17 +175,20 @@ def __init__(
...
 
         self._event_pattern = EventPattern(
             source=["aws.s3"],
-            detail={
-                "eventSource": ["s3.amazonaws.com"],
-                "eventName": ["PutObject"],
-                "requestParameters": request_parameters,
-            },
+            detail=event_detail,
+            detail_type=[],
```
Comment (jaidisido) [inline core/aws_ddk_core/stages/kinesis_s3.py:191]: And this be `["Object Created"]`?

**Review by jaidisido — CHANGES_REQUESTED** [review]

File: core/aws_ddk_core/stages/s3_event.py (hunk 2 above)
Comment (kukushking) [reply core/aws_ddk_core/stages/s3_event.py:77]: Oops

File: core/aws_ddk_core/stages/kinesis_s3.py (hunk 3 above)
Comment (kukushking) [reply core/aws_ddk_core/stages/kinesis_s3.py:191]: Thanks

**Review by jaidisido — APPROVED** [review]

**Pull request merged**